game/src/info/trip.rs

```diff
@@ -386,6 +389,329 @@ pub fn cancelled(ctx: &mut EventCtx, app: &App, id: TripID) -> Widget {
+                .render(ctx)
+                .centered_on(intersection.polygon.center());
+            details.unzoomed.append(unzoomed_rendered.clone());
+            details.zoomed.append(zoomed_rendered);
+        }
+    }
+}
+
+/// Highlights lanes which were "slow" on the map
+fn highlight_slow_lanes(app: &App, details: &mut Details, id: TripID, ctx: &EventCtx) {
+    let data_opt = &app.primary.sim.get_analytics().trip_lane_speeds(id);
+    if data_opt.is_some() {
+        let data = data_opt.unwrap();
+        for (id, speed_percent) in data {
+            let lane = app.primary.map.get_l(*id);
+            let (fg_color, bg_color) = if speed_percent > &95 {
+                (Color::WHITE, app.cs.normal_slow_intersection)
+            } else if speed_percent > &60 {
+                (Color::BLACK, app.cs.slow_intersection)
+            } else {
+                (Color::WHITE, app.cs.very_slow_intersection)
+            };
+            details.unzoomed.push(
+                Fill::Color(bg_color),
+                lane.lane_center_pts.make_polygons(Distance::meters(10.0)),
+            );
+            details.zoomed.extend(
+                Fill::Color(bg_color),
+                lane.lane_center_pts.dashed_lines(
+                    Distance::meters(0.75),
+                    Distance::meters(1.0),
+                    Distance::meters(0.4),
+                ),
+            );
+            // Draw label after, to prevent being obscured by the line
+            let unzoomed_label = Text::from(TextSpan::from(
+                Line(format!("{}s", speed_percent)).body().fg(fg_color),
+            ))
+            .bg(bg_color);
+            let zoomed_label = Text::from(TextSpan::from(
+                Line(format!("{}s", speed_percent)).small().fg(fg_color),
+            ))
+            .bg(bg_color);
+
+            let (pt, _) = lane
+                .lane_center_pts
+                .must_dist_along(lane.lane_center_pts.length() / 2.0);
+            let unzoomed_rendered = unzoomed_label.render(ctx).centered_on(pt); //.rotate(angle);
+            let zoomed_rendered = zoomed_label.render(ctx).centered_on(pt); //.rotate(angle);
+            details.unzoomed.append(unzoomed_rendered.clone());
+            details.zoomed.append(zoomed_rendered);
+        }
+    }
+}
+
+/// Helper func for make_bar()
+/// Builds a default text overlay widget
+fn build_text(msgs: &Vec<String>, distance: &Distance) -> Text {
+    let mut display_txt = Text::new();
+    for msg in msgs {
+        display_txt.add(Line(msg));
+    }
+    display_txt.add(Line(format!(
+        "  Distance covered: {}",
+        distance.describe_rounded()
+    )));
+    display_txt
+}
+
+/// Makes the timeline bar in trip info panel
+fn make_bar(
+    ctx: &mut EventCtx,
+    app: &App,
+    trip_id: TripID,
+    phases: &Vec<TripPhase>,
+    progress_along_path: Option<f64>,
+) -> Widget {
+    let map = &app.primary.map;
+    let data_inter_opt = &app
```

Comment:
Style: This variable name is confusing, how about `intersection_delays`? I don't tend to mark optional stuff in the name itself, since the compiler will catch problems, and sometimes I'll do `if let Some(intersection_delays) = intersection_delays { ... }` to avoid picking two names.

Reply:
Wow, I didn't know you could do that with if lets!
Changed